code/modules/antagonists/changeling/powers/fakedeath.dm

```diff
@@ -20,22 +20,27 @@
 	if(revive_ready)
 		INVOKE_ASYNC(src, PROC_REF(revive), user)
 		disable_revive(user) // this should be already called via signal, but just incase something wacky happens
+		return TRUE
 
-	else if(enable_fakedeath(user))
-		to_chat(user, span_changeling("We begin our stasis, preparing energy to arise once more."))
+	var/death_duration_mod = 1
+	if(user.has_status_effect(/datum/status_effect/gutted))
+		death_duration_mod *= 8 // Anti-megafauna cheese
 
-	else
-		stack_trace("Changeling revive failed to enter fakedeath when it should have been in a valid state to.")
+	if(enable_fakedeath(user, duration_modifier = death_duration_mod))
+		to_chat(user, span_changeling("We begin our stasis, preparing energy to arise once more."))
+		if(death_duration_mod > 1)
+			to_chat(user, span_changeling("<b>Our body has sustained severe damage, and will take longer to regenerate.</b>"))
+		return TRUE
 
-	return TRUE
+	stack_trace("Changeling revive failed to enter fakedeath when it should have been in a valid state to.")
```

Comment:
turn this into an early return instead? also we really should be explicitly returning false or something instead of `null`

Reply:
like invert the check to be `(if(!enable_fakedeath(args)); stack_trace("fuck"); return FALSE` so the more important chunk of code that actually does work doesn't have to stick out like that